ably/rest_channel.go

```diff
@@ -46,17 +49,26 @@ func newRestChannel(name string, client *REST) *RestChannel {
 	return c
 }
 
-func (c *RestChannel) Publish(name string, data interface{}) error {
-	messages := []*proto.Message{
+// Publish publishes a message on the channel.
+func (c *RESTChannel) Publish(ctx context.Context, name string, data interface{}) error {
+	return c.PublishAll(ctx, []*Message{
 		{Name: name, Data: data},
-	}
-	return c.PublishAll(messages)
+	}, nil)
 }
 
-// PublishAll sends multiple messages in the same http call.
-// This is the more efficient way of transmitting a batch of messages
-// using the Rest API.
-func (c *RestChannel) PublishAll(messages []*proto.Message) error {
+// Message is what Ably channels send and receive.
+type Message = proto.Message
+
+// PublishAll publishes multiple messages in a batch.
+//
+// The params, if any, will be set as additional query parameters in the
+// resulting HTTP request to the REST API.
+func (c *RESTChannel) PublishAll(ctx context.Context, messages []*Message, params map[string]string) error {
```

Comment:
Given `params` is optional, then as per https://github.com/ably/ably-go/issues/197#issuecomment-705526229 should there be a separate `PublishAllWithOptions` (or `PublishBatchWithOptions`), and a `WithPublishParam` option?

Reply:
Sure, in fact this is what prompted the discussion but then I forgot to actually do it before submitting the PR. Done at d987390.

Reply:
Great 👍